src/Microsoft.IdentityModel.Tokens/SymmetricSignatureProvider.cs

```diff
@@ -77,7 +77,15 @@ public SymmetricSignatureProvider(SymmetricSecurityKey key, string algorithm)
 
             try
             {
-                _keyedHash.Key = key.Key;
+                SymmetricSecurityKey symmetricSecurityKey = key as SymmetricSecurityKey;
```

Comment:
Need error checking on SecurityKey type, make sure key material (JsonWebKey.K) is not null. 
We can pass key into GetKeyedHashAlgorithm and new up the HMACxxx with the key.